pyigm/clustering/xcorr_utils.py

```diff
@@ -322,7 +386,175 @@ def spline_sensitivity(galreal, Nmin=20, magmin=17., magmax=26., delta_mag=0.5,
+    fl:      numpy array of normalized flux.
+    er:      numpy array of error in the normalized flux of the spectrum for
+             a given wavelength.
+    sl:      significance level for the detection of the absorption line.
+    R:       resolution of the spectrograph, assumed constant
+    FWHM:    Full-width at half maximum in pixels (assumed constant). This
+             parameter defines the smoothing scale for Wmin.
+    ion:     Name of the ion. Function only valid for HI so far.
+
+    From the error we calculate the Wmin = sl * wa * er / (1+z) / R,
+    where z = wa/w0 - 1 (w0 is the rest frame wavelenght of the
+    transition) and R is the resolution of the spectrograph. We then
+    smooth Wmin with a boxcar along FWHM pixels.
+
+    For the given absorber we transform (logN_obs,b_obs) to a W_obs assuming
+    linear part of the curve-of-growth.
+
+    We then compute the redshifts where W_obs could have been observed
+    according to the given Wmin, and place Nrand new absorbers with
+    the same properties as the given one accordingly.
+    """
+    from linetools.analysis.absline import Wr_from_N_b_transition
+    from linetools.lists.linelist import LineList
+    ism = LineList('ISM')
+
+    Nrand = int(Nrand)
+    absrand = absreal.repeat(Nrand)
+
+    zmin = np.min(absreal.ZABS)
+    zmax = np.max(absreal.ZABS)
+
+    if ion == 'HI':
+        z_Lya, Wmin_Lya = compute_Wmin(wa, fl, er, sl=sl, R=R, FWHM=FWHM, ion='HI')
+        z_Lyb, Wmin_Lyb = compute_Wmin(wa, fl, er, sl=sl, R=R, FWHM=FWHM, ion='HILyb')
+
+    for i in range(len(absreal)):
+
+        if absreal.ZABS[i] > np.max(z_Lya):  # lines that were observed through Lyb
+            Wr = Wr_from_N_b_transition(absreal.LOGN[i], absreal.B[i], 'HI 1025', linelist=ism)
+            #Wr = logN_b_to_Wr(absreal.LOGN[i], absreal.B[i], ion='HILyb')
+            z = z_Lyb
+            z = np.where(z <= z_Lya, -1., z)  # mask out region with Lya coverage
+            Wmin = Wmin_Lyb
+        else:  # lines that were observed through Lya only
+            Wr = Wr_from_N_b_transition(absreal.LOGN[i], absreal.B[i], 'HI 1215', linelist=ism)
+            #Wr = logN_b_to_Wr(absreal.LOGN[i], absreal.B[i], ion='HI')
+            z = z_Lya
+            Wmin = Wmin_Lya
+
+        zgood = (Wr > Wmin) & (z >= zmin) & (z < zmax)
+
+        if np.sum(zgood) == 0:
+            from matplotlib import pyplot as plt
+            plt.plot(z, Wmin, drawstyle='steps-mid')
+            plt.axis([z[0], z[-1], 0, 0.1])
+            plt.show()
+            print(Wmin)
+        assert np.sum(zgood) > 0, \
+            'There are not regions in the spectrum with Wmin<{} A. The minimum is {}. Adjust significance.'.format(
+                Wr, np.min(Wmin))
+
+        # Random time
+        rand_z = RanDist(z, zgood * 1.)
+        zrand = rand_z.random(Nrand)
+        absrand.ZABS[i * Nrand:(i + 1) * Nrand] = zrand
+
+    return absrand
+
+def mask_galactic():
```

Comment:
change name to `get_galactic`, as this function is not masking.

Reply:
I wonder whether a function like mask_galactic() could be incorporated into a more general utility module of pyigm.

Reply:
edited;
maybe